libraries/AP_RangeFinder/AP_RangeFinder_BLPing.cpp

```diff
@@ -66,12 +36,65 @@ void AP_RangeFinder_BLPing::update(void)
 
 void AP_RangeFinder_BLPing::init_sensor()
 {
-    // request distance from sensor
-    send_message(BLDPIN_MSGID_DISTANCE_SIMPLE, nullptr, 0);
+    // Set message interval between pings in ms
+    uint16_t ping_interval = _sensor_rate_ms;
+    protocol.send_message(uart, PingProtocol::MessageId::SET_PING_INTERVAL, reinterpret_cast<uint8_t*>(&ping_interval), sizeof(ping_interval));
```

Comment:
We should read back the setting to confirm it.

Reply:
The patch follows the old implementation, it only moves to continuous request and break the protocol in a different class. We can improve it to make it totally compatible with the protocol in future patchs.